### PR TITLE
734 use decimal for money

### PR DESCRIPTION
**Pull request by AndrewYHuang**
Updated BankAccount, BookStore and Ledger exercises to use decimal for money instead of float or double.
Updated RenderNumber and Render with decimal rendering.
Updated BookStore test generator to use decimal literals for BookStore tests.

### CODE REVIEW

File: generators/Output/Rendering/RenderNumber.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@ namespace Exercism.CSharp.Output.Rendering
 {
     public partial class Render
     {
+        public string Decimal(decimal dec) => $"{dec.ToString(CultureInfo.InvariantCulture)}m";
```
Comment (ErikSchierboom) [inline generators/Output/Rendering/RenderNumber.cs]: Could you add a blank line after this method?

File: exercises/ledger/Ledger.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 public class LedgerEntry
 {
-   public LedgerEntry(DateTime date, string desc, float chg)
+   public LedgerEntry(DateTime date, string desc, decimal chg)
```
Comment (ErikSchierboom) [inline exercises/ledger/Ledger.cs]: Nice! Could you also update the `Example.cs` implementation (which is in the same directory)?

File: exercises/book-store/BookStore.cs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 public static class BookStore
 {
-    public static double Total(IEnumerable<int> books)
+    public static decimal Total(IEnumerable<int> books)
```
Comment (ErikSchierboom) [inline exercises/book-store/BookStore.cs]: Nice! Could you also update the `Example.cs` implementation (which is in the same directory)?

File: exercises/bank-account/BankAccount.cs [hunk 4]
```diff
@@ -12,15 +12,15 @@ public void Close()
         throw new NotImplementedException("You need to implement this function.");
     }
 
-    public float Balance
+    public decimal Balance
```
Comment (ErikSchierboom) [inline exercises/bank-account/BankAccount.cs]: Nice! Could you also update the `Example.cs` implementation (which is in the same directory)?

**Comment by AndrewYHuang** [comment]
Regarding lines 942-971 of the ci log, does the example implementation need updating? c# really doesn't like doing equals(decimal, float)

File: exercises/bank-account/BankAccount.cs (hunk 4 above)
Comment (AndrewYHuang) [reply exercises/bank-account/BankAccount.cs]: Ok, sure! Didn't see that

**Comment by ErikSchierboom** [comment]
> does the example implementation need updating

It does indeed.

**Comment by AndrewYHuang** [comment]
Made all the requested changes: Updated examples and added that blank line to RenderNumber

**Comment by ErikSchierboom** [comment]
Merged! This is great work. Thanks a lot for doing this. 🎉

**Pull request merged**